lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2468,6 +2468,21 @@ static LogicalResult verifyHWStructCastOp(HWStructCastOp cast) {
   return success();
 }
 
+static LogicalResult verifyBitCastOp(BitCastOp cast) {
+
+  auto inTypeBits = getBitWidth(cast.getOperand().getType().cast<FIRRTLType>());
+  auto resTypeBits = getBitWidth(cast.getType().cast<FIRRTLType>());
+  StringRef errorMsg =
+      "the input and result must have known non-zero bit widths";
```

Comment:
I'd recommend duplicating the cast.emitError to simplify the logic.  No need for an errorMsg variable.